app/controllers/submissions_controller.rb

```diff
@@ -0,0 +1,77 @@
+class SubmissionsController < ApplicationController
+  before_action :authenticate_user!, only: [:create, :create_comment]
+  before_action :authenticate_to_submit, only: :new
+
+  helper_method :posts, :post, :new_post, :new_comment
+
+  def new
+  end
+
+  def show
+  end
+
+  def index
+  end
+
+  def create
+    @post = Submission.new(post_params)
+    @post.user_id = current_user.id
+    if @post.save
+      flash[:alert] = "Your submission was saved"
+      redirect_to submission_path(@post.id)
+    else
+      flash[:alert] = @post.errors.messages
+      redirect_to new_submission_path
+    end
+  end
+
+  def create_comment
+    @comment = Submission.new(comment_params)
+    @comment.user_id = current_user.id
+    @comment.post_id = parent_post_id
+    if @comment.save
+      flash[:alert] = "Your submission was saved"
+      redirect_to submission_path(@comment.post_id)
+    else
+      flash[:alert] = @comment.errors.messages
+      redirect_to submission_path(parent_post_id)
+    end
+  end
+
+  private
+
+  def authenticate_to_submit
+    if !user_signed_in?
+      flash[:alert] = "You must be logged in to submit a post"
+      redirect_to new_user_session_path
+    end
+  end
+
+  def posts
+    @posts = Submission.where(post_id: nil)
+  end
+
+  def post
+    @post = Submission.find(params[:id])
+  end
+
+  def new_post
+    @post = Submission.new
+  end
+
+  def new_comment
+    @comment = Submission.new
```

Comment:
Should these be cached, like `@comment ||= Submission.new`, so they do not create new records on every call?

Reply:
Thanks @nhunzaker, I've been stuck on this because I thought the point of `.new` is that a new object will be instantiated but it's not saved with AR until `.save` is applied. Still, I guess that new object is still taking up space in the app's working memory.